### PR TITLE
Docs update

### LINKED ISSUE #716: Documentation topics should be reorganised.
### HydroMT version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://github.com/Deltares/hydromt)

### Kind of issue

Docs are unclear

### Location of the documentation

_No response_

### Documentation problem

Currently the documentation is very centered around the `build` `update` and other top level CLI actions. We're currently adding more functionallity that doesn't have a good home under this rubric yet, so this should be adressed. 

### Suggested fix for documentation

 - [x] beginner
 - [ ] advanced
 - [ ] plugin

### LINKED ISSUE #977: update migration guide section on API
### HydroMT version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://github.com/Deltares/hydromt)

### Kind of issue

Docs are wrong

### Location of the documentation

dev/migrating-to-v1.rst

### Documentation problem

api section needs updating after #959 

### Suggested fix for documentation

_No response_

### PR DESCRIPTION
**Pull request by savente93**
## Issue addressed
Fixes #716 
Fixes #977 

## Explanation
I didn't actually update any of the text, but given that diffs can become a little hard to follow through reorganisations I thought I'd propose the new structure in a separate PR. Once this get merged I'll get started on updating the sections.  
## General Checklist
- [ ] Updated tests or added new tests
- [ ] Branch is up to date with `main`
- [ ] Tests & pre-commit hooks pass
- [ ] Updated documentation
- [ ] Updated changelog.rst

## Data/Catalog checklist
- [ ] `data/catalogs/predefined_catalogs.yml` has not been modified.
- [ ] None of the old `data_catalog.yml` files have been chagned
- [ ] `data/chagnelog.rst` has been updated
- [ ] new file uses `LF` line endings (done automatically if you used `update_versions.py`)
- [ ] New file has been tested locally
- [ ] Tests have been added using the new file in the test suite

## Additional Notes (optional)
Add any additional notes or information that may be helpful.

### CODE REVIEW

File: docs/guides/user_guide/index.rst [hunk 1]
```diff
@@ -0,0 +1,21 @@
+Basic user guide
```
Comment (Jaapel) [inline docs/guides/user_guide/index.rst:1]:
```suggestion
User Guide
```

File: pixi.toml [hunk 2]
```diff
@@ -171,6 +171,7 @@ xarray = "*"
 xmltodict = "*"
 zarr = "*"
 xugrid = ">=0.9.0"
+sphinx-autoapi = ">=3.0.0,<3.1"
```
Comment (Jaapel) [inline pixi.toml:174]: Did we not try and discard this option?

File: pyproject.toml [hunk 3]
```diff
@@ -91,6 +91,7 @@ doc = [
   "sphinx_autosummary_accessors", # doc layout
   "sphinx",                       # build docks
   "sphinx_design",                # doc layout
+  "sphinx-autoapi",               # auto api documentation
```
Comment (Jaapel) [inline pyproject.toml:94]: Did we not try and discard this option?

**Review by Jaapel — CHANGES_REQUESTED** [review]
Some minor comments

File: pyproject.toml (hunk 3 above)
Comment (savente93) [reply pyproject.toml:94]: we did, seems like I forgot to remove it. my bad

File: docs/conf.py [hunk 4]
```diff
@@ -163,7 +163,7 @@ def clean_str(s):
     "sphinx.ext.autodoc",
     "sphinx.ext.viewcode",
     "sphinx.ext.todo",
-    "sphinx.ext.napoleon",
+'sphinx.ext.napoleon',
```
[the remark this answers is not in the document]
Comment (savente93) [reply docs/conf.py:166]: it is indeed

File: docs/guides/plugin_dev/best_practices.rst [hunk 5]
```diff
@@ -0,0 +1,4 @@
+.. _best_practices:
+
+Best Practices
+===================
```
[the remark this answers is not in the document]
Comment (savente93) [reply docs/guides/plugin_dev/best_practices.rst:4]: I added a document where we can describe this in better in a new PR

**Review by Jaapel — APPROVED** [review]
👍
